### PR TITLE
Compatibility pass on flutter/physics tests for JavaScript compilation. (6)

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Flags certain test cases in rendering using package:test tags in order to facilitate landing support for running all framework tests in a browser.

Requires #33349

### CODE REVIEW

File: packages/flutter/test/physics/newton_test.dart [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+@TestOn('!chrome') // different numeric behavior.
```
Comment (Hixie) [inline packages/flutter/test/physics/newton_test.dart]: this seems very distressing. this code is being used on web, we can't just ignore the differences.

Comment (yjbanov) [reply packages/flutter/test/physics/newton_test.dart]: Let's change this to `Tags('web_unimplemented')` and treat it as a bug.

Comment (yjbanov) [reply packages/flutter/test/physics/newton_test.dart]: We have not yet seen differences in how double math is implemented between JS and Dart. We should see identical results here.

Comment (jonahwilliams) [reply packages/flutter/test/physics/newton_test.dart]: Agreed we should treat this as a bug. In this case, the rounding is off - something like `8` vs `7.999999999`.

**Comment by jonahwilliams** [comment]
Updated to remove skip, adding a `closeTo` matcher instead, and to run on CI

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**